src/negotiate.rs

```diff
@@ -40,34 +40,34 @@ pub struct NegotiateConfig {
     pub protocol_config: Box<dyn ProtocolConfig>,
     pub package_list: Option<String>,
     pub hostname: String,
-    pub network_client_factory: Box<dyn NetworkClientFactory>,
+    pub yield_point: Option<YieldPointLocal>
```

Comment:
No yield point should be held by structures

Reply:
udpated